LibationWinForms/ProductsGrid.cs

```diff
@@ -149,10 +152,10 @@ public void Display()
 
 		public void RefreshRow(string productId)
 		{
-			var rowIndex = getRowIndex((ge) => ge.AudibleProductId == productId);
+			var liveGridEntry = getGridEntry((ge) => ge.AudibleProductId == productId);
 
-			// update cells incl Liberate button text
-			_dataGridView.InvalidateRow(rowIndex);
```

Comment:
Where is the row invalidated/re-drawn now?

Reply:
GridEntry implements [INotifyPropertyChanged](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged?view=net-5.0). When added to a BindidingList, the list subscribes to the PropertyChanged event in every GridEntry. Firing the [PropertyChanged](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged.propertychanged?view=net-5.0) triggers the DataGridView to update the view for GridEntry. If a property name is specified (as it is below), DataGridView will only update the cell bound to that property. Since RefreshRow is only used to update display of the Liberate column, we only need to notify that the Liberate property has changed, and DataGridView will redraw it. If you want to notify all properties in the GridEntry, don't specify a property name.

Reply:
RefreshRow might more accurately be named RefreshLiberate.